treerec/tests/test_individual_output.py

```diff
@@ -0,0 +1,73 @@
+            pedigree_id = int(fields[1])
+            if pedigree_id in slim and slim[pedigree_id].type == "retain":
+                # We have a duplicate: "remember"ed takes priority
+                continue
+            slim[pedigree_id] = SLiMindividual(
+                type=store,
+                population=int(fields[2]),
+                pos=[float(p) for p in fields[3].split(",")],
+                nodes=[int(p) for p in fields[4].split(",")],
+            )
+        return slim
+
+    def test_ts_slim_consistency(self):
+        # load tree sequence
+        for ts in self.get_ts():
+            ts = ts.simplify(filter_individuals=True)  # check we are fully simplified
+            # this is a dictionary of SLiM -> tskit ID (from metadata in nodes)
+            ids = self.get_slim_ids(ts)
+            # this is a dict of tskit ID -> index in samples
+            slim_individuals = self.read_test_individual_output(
+                filename="test_output/slim_individual_output.txt")
+            # check that all the individual data lines up
+            ts_individuals = {i.metadata['pedigree_id']: i for i in ts.individuals()}
+            num_expected = 0
+            for ped_id, slim_ind in slim_individuals.items():
+                if slim_ind.type == "retain":
+                    # retained individuals should only be present in the TS if they
+                    # have genomic material. We should check which have nodes that exist
+                    ts_nodes = [ids[n] for n in slim_ind.nodes if n in ids]
+                    if len(ts_nodes) > 0:
+                        num_expected += 1
+                        assert ped_id in ts_individuals
+                        metadata = ts_individuals[ped_id].metadata
+                        assert slim_ind.population == metadata['subpopulation']
+                        assert np.allclose(slim_ind.pos, ts_individuals[ped_id].location)
+                        for n in ts_nodes:
+                            assert not ts.node(n).is_sample()
+                    else:
+                        assert ped_id not in ts_individuals
+                else:
+                    # permanently "remember"ed individuals and ones that exist during ts
+                    # output are always present, and their nodes should be marked as
+                    # as samples
+                    num_expected += 1
+                    assert ped_id in ts_individuals
+                    metadata = ts_individuals[ped_id].metadata
+                    assert slim_ind.population == metadata['subpopulation']
+                    assert np.allclose(slim_ind.pos, ts_individuals[ped_id].location)
+                    ts_nodes = set([ids[n] for n in slim_ind.nodes])
+                    assert ts_nodes == set(ts_individuals[ped_id].nodes)
+                    for n in ts_nodes:
+                        assert ts.node(n).is_sample()
+                        if ts.node(n).time > 0:
```

Comment:
For a nonWF model this won't necessarily be true. You could instead do like
```
alive = ts.individuals_alive_at(0)
if ped_id not in alive:
   assert slim_ind.type == "remember"
```
(if you make `ts` a SlimTreeSequence)

Hm, and we should definitely add a nonWF test for this stuff, too, if there isn't one.

Reply:
Oh, nice spot. I should add another recipe I guess.

Reply:
Added a non-WF recipe and corrected the tests